tests/rptest/archival/abs_client.py

```diff
@@ -72,15 +72,20 @@ def delete_bucket(self, name: str):
         blob_service = BlobServiceClient.from_connection_string(self.conn_str)
         blob_service.delete_container(name)
 
-    def empty_bucket(self, name: str, parallel=False):
-        # TODO: implement `parallel` as/when we start running any scale tests
-        # on Azure
+    def empty_bucket(self, name: str, parallel=True):
```

Comment:
In the S3 client, parallel deletion means batching (into DeleteObjects) and also parallel execution (so we're running multiple DeleteObjects in parallel).

I wonder if we could generalize it into something that takes a batch size, a list lambda and a delete lambda, and call the same code from both clients.